llvm/lib/Target/X86/X86AsmPrinter.cpp

```diff
@@ -692,13 +692,32 @@ static bool printAsmVRegister(const MachineOperand &MO, char Mode,
   return false;
 }
 
+void X86AsmPrinter::diagnoseAsmOperandError(LLVMContext &C,
+                                            const AsmOperandErrorCode EC,
+                                            const char *AsmStr, uint64_t Loc) {
+  AsmPrinter::diagnoseAsmOperandError(C, EC, AsmStr, Loc);
+  std::string msg;
+  raw_string_ostream Msg(msg);
+  switch (EC) {
+  default:
+    break;
+  case AsmOperandErrorCode::CONSTRAINT_H_ERROR:
```

Comment:
I don't see where the X86 backend ever returns this

Reply:
I did not understand comment.
From Here, we are first printing some generic diagnostic present without this patch. e,g, regarding "invalid Operand with Src Location" , Since target can have constraint code specific to architecture, I used specialized `X86AsmPrinter::diagnoseAsmOperandError` to just provide helpful diagnostics for 'H;

Reply:
But H doesn't return this error on X86, so this is dead code. Which you would know if you wrote a test for this code (something else that's lacking from this PR).